pkg/crypto/crypto_test.go

```diff
@@ -84,27 +84,91 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 	//The keys are base64 encoded. Decode it before passing to encryption function
 	keyDecoded, err := b64.StdEncoding.DecodeString(string(key))
 	if err != nil {
-		panic(err.Error())
+		return(err.Error()) 
 	}
 	nonceDecoded, err := b64.StdEncoding.DecodeString(string(nonce))
 	if err != nil {
-		panic(err.Error())
+		return(err.Error())
 	}
 
 	block, err := aes.NewCipher(keyDecoded)
 	if err != nil {
-		panic(err.Error())
+		return(err.Error())
 	}
 
 	aesgcm, err := cipher.NewGCM(block)
 	if err != nil {
-		panic(err.Error())
+		return(err.Error())
 	}
 
 	plaintextBytes, err := aesgcm.Open(nil, nonceDecoded, sDec, nil)
 	if err != nil {
-		panic(err.Error())
+		return(err.Error())
 	}
 
 	return string(plaintextBytes)
 }
+
+func TestDecryptConfigMapWrongkey(t *testing.T)  {
+        
+	testString := "Sample String to Test Symmetric Encryption"
+
+        dir1, err := ioutil.TempDir("", "TestEncryption-2")
+        if err != nil {
+                t.Error(err)
+        }
+        defer os.RemoveAll(dir1)
+
+        keyPath := dir1 + "symmKeyFile"
+        wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+        key := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, key); err != nil {
+                t.Error(err)
+        }
+        wrongkey := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the Key
+        keyEnc := b64.StdEncoding.EncodeToString(key)
+        //encode wrong key
+        wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+
+        err = ioutil.WriteFile(keyPath, []byte(keyEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+        err=ioutil.WriteFile(wrongkeyPath, []byte(wrongkeyEnc), 0644)
+	if err != nil {
```

Comment:
need to run gofmt => `gofmt -s -w <file>`

Reply:
> 
> 
> need to run gofmt => `gofmt -s -w <file>`
Thanks for the suggestions. done now.